File: backend/schemas/query.py
from typing import Any, Optional

from pydantic import BaseModel, Field


class QueryRequest(BaseModel):
    connection_id: str
    sql: str
    params: Optional[list[Any]] = Field(default_factory=list)


class QueryResult(BaseModel):
    columns: list[str]
    rows: list[dict[str, Any]]
    row_count: int
    execution_ms: int

File: frontend/src/types/query.ts
export interface QueryRequest {
  connection_id: string;
  sql: string;
  params?: unknown[];
}

export interface QueryResult {
  columns: string[];
  rows: Record<string, unknown>[];
  row_count: number;
  execution_ms: number;
}

export interface PromptRequest {
  prompt: string;
  connection_id?: string;
  dashboard_id?: string;
}

export interface QueryInfo {
  sql: string;
  params: unknown[];
  execution_ms: number;
  row_count: number;
}

export interface WidgetResult {
  id?: string;
  type: string;
  title: string;
  prompt_used: string;
  query_config: Record<string, unknown>;
  chart_config: Record<string, unknown>;
  layout_position: Record<string, unknown>;
  data: Record<string, unknown>[];
  explanation: string;
}

export interface PromptResponse {
  widget: WidgetResult;
  query_info: QueryInfo;
  explanation: string;
}
